Repository is now instanciated on the fly

abienvenu opened the pull request:
No more repository instantiated in constructor of ClientManager and TokenManager.

This avoids a database connection to be established for every request.

Fixes issue https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/422

ruudk reviewed (approved):
:+1:

ruudk commented:
Tests are failing tho

abienvenu commented:
Right, there was a test conflicting with the very purpose of this pull request. I removed this test. There are still some failing tests, but I guess these out of memory errors are unrelated.

dinamic commented:
The unit tests have improved greatly the last couple of weeks.

Could you rebase on top of master?

abienvenu committed:
Onthefly repo for ClientManager and TokenManager

This avoids a database connection to be established for every request
Fixes issue https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/422

abienvenu commented:
Ok, I made the rebase and tests are fine now.

dkarlovi commented:
Can you handle authcodemanagers too?

abienvenu committed:
Onthefly repo for AuthCodeManager

This avoids a database connection to be established for every request
Completes fix for issue https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/issues/422

abienvenu commented:
Yes @dkarlovi, good point. AuthCodeManagers are handled as well now.
